docs/product/visualizations.md

```diff
@@ -251,19 +252,27 @@ In particular:
 Very important information is how visualization architecture works to make them
 interactive and fast. Whenever new data is computed by the compiler and
 visualization is attached to it, it is sent to GUI to be displayed. However,
-sending really big chunks of data will kill the performance. When defining a
+sending huge chunks of data will kill the performance. When defining a
 visualization user is capable of defining a chunk of Enso code (as a string).
 This code is part of the visualization definition and is stored server-side.
-Visualizations are allowed to change the code at runtime. This code defines an
+Visualizations are allowed to change the code at runtime (in JavaScript 
+visualization you may use the `setPreprocessor` method). This code defines an
 Enso function, which will be run by the compiler on data the visualization is
-attached to. Only the results of this code will be sent to the GUI. For example,
-imagine you want to display a heatmap of 10 million points on a map. And these
-points change rapidly. Sending such amount of information via WebSocket could be
-too much, and you (as the visualization author) might decide that the
-visualization image should be generated on the server, and your visualization
-is meant only to display the resulting image. In such a scenario, you can define
-in your visualization an Enso function which will compute the image on the
-server!
+attached to. Only the results of this code will be sent to the GUI. In case 
+of JSON input format the result of the call should be a valid JSON string. The
+code will be evaluated in context of the `Main` module in the project where 
+visualization is defined - you may use any symbol defined or imported in 
+that module.
+
+For example, imagine you want to display a heatmap of 10 million points on 
+a map, and these points change rapidly. Sending such amount of information via
+WebSocket could be too much, and you (as the visualization author) might decide 
+that the visualization image should be generated on the server, and your 
+visualization is meant only to display the resulting image. In such a scenario, 
+you can define in your visualization an Enso function which will compute the 
+image on the server!
+
+
```

Comment:
I love the example in the docs!

This has a few grammar mistakes (like the rest of the docs here) - would you be so nice and log-in to `https://app.grammarly.com/` and copy-paste this doc there? It will show all errors with auto-fix buttons. In fact, we would use Grammarly for all comments. Its so bad its not integrated with intellij